### PR TITLE
Fix Tracks related Pixel quantities at HLT DQM (backport)

### PR DESCRIPTION
**Pull request by arossi83**
#### PR description:

Monitor of the tracks related Pixel quantities at HLT DQM never works and it was disabled at some point, with this PR we introduce the changes needed (tracks refitting) in order to be able to monitor Pixel Tracks quantities at HLT DQM.
In this PR we also update the collection to be used by the Standard DQM Pixel Residuals Monitor in order to use the refitted tracks.

#### PR validation:

runTheMatrix.sh -l 140.063
Manually run the Online HLT client at P5 with old stored DQM stream file

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Backport of #40873

### CODE REVIEW

**Comment by mmusich** [comment]
test parameters:
   * workflow = 140.063

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
urgent
   * please @cms-sw/dqm-l2 consider this for integration in 13_0_0.

**Comment by mmusich** [comment]
@arossi83 the uni test `TestDQMOnlineClient-hlt_dqm_sourceclient` is [failing](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cc786b/30879/unitTests/src/DQM/Integration/test/TestDQMOnlineClient-hlt_dqm_sourceclient/testing.log) with  
```display
----- Begin Fatal Exception 24-Feb-2023 12:25:10 CET-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 355380 lumi: 19 event: 26716348 stream: 0
   [1] Running path 'hlt4vector'
   [2] Calling method for module SiPixelClusterShapeCacheProducer/'hltSiPixelClusterShapeCache'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: edmNew::DetSetVector<SiPixelCluster>
Looking for module label: hltSiPixelClusters
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```

this happens because the input file for the test `/store/express/Run2022B/ExpressPhysics/FEVT/Express-v1/000/355/380/00000/2b3f408f-883a-4f64-8083-9e5786589069.root` is a FEVT file from StreamExpress and as such it doesn't have the necessary HLT collections. 
The test has been running successfully in IB and PR tests so far only because all the (DQM module) consumers in its path are apparently well protected in terms of missing inputs, while `SiPixelClusterShapeCacheProducer` is not. 
I think the right move would be to change the input of this unit test to use a file that instead *does* contain the HLT-produced collections, since otherwise the test is not testing anything, but I am not sure if there's any readily available such file. A `FEVTHLTALL` file from `/HLTMonitor/Run2022*Express-v*/FEVTHLTALL` would serve the purpose but I cannot find any on disk. 
@cms-sw/hlt-l2 FYI.

**Comment by missirol** [comment]
Interesting.. I couldn't find any files from `/HLTMonitor/Run2022*Express-v*/FEVTHLTALL` either.

I could only find [1]; that might help for local tests, but I guess we should not rely on this non-standard path in CMSSW. _Maybe_, one quick workaround could be asking to copy this in a proper folder of the cmsbot cache (then CMSSW tests would see it). Not sure how to proceed.
[1]
```
/eos/cms/tier0/store/backfill/1/express/Tier0_REPLAY_2022/HLTMonitor/FEVTHLTALL/Express-v710/000/356/005/00000/0f207c66-8cda-4c3d-b4c8-3b063efc72e8.root
```

File: DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_cff.py [hunk 1]
```diff
@@ -2,8 +2,15 @@
...
+
+hltSiPixelClusterShapeCache = siPixelClusterShapeCache.clone(src = 'hltSiPixelClusters')
+hltrefittedForPixelDQM = refittedForPixelDQM.clone(src ='hltMergedTracks')
 
 sipixelMonitorHLTsequence = cms.Sequence(
-    hltSiPixelPhase1ClustersAnalyzer
-    #+ hltSiPixelPhase1TrackClustersAnalyzer
+    hltSiPixelClusterShapeCache
+    + hltSiPixelPhase1ClustersAnalyzer
+    + refittedForPixelDQM
```
Comment (mmusich) [inline DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_cff.py:14]: shouldn't this be

```suggestion
    + hltrefittedForPixelDQM
```

?

**Comment by mmusich** [comment]
so this patch:
```diff
diff --git a/DQM/Integration/python/clients/hlt_dqm_sourceclient-live_cfg.py b/DQM/Integration/python/clients/hlt_dqm_sourceclient-live_cfg.py
index 3e6b28ecb1f..cc04e270abf 100644
--- a/DQM/Integration/python/clients/hlt_dqm_sourceclient-live_cfg.py
+++ b/DQM/Integration/python/clients/hlt_dqm_sourceclient-live_cfg.py
@@ -46,8 +46,8 @@ process.ClusterShapeHitFilterESProducer = cms.ESProducer( "ClusterShapeHitFilter
 )
 #SiStrip Local Reco
 process.load("CalibTracker.SiStripCommon.TkDetMapESProducer_cfi")
-#SiPixelTemplate
-process.load("CalibTracker.SiPixelESProducers.SiPixelTemplateDBObjectESProducer_cfi")
+#Track refitters
+process.load("RecoTracker.TrackProducer.TrackRefitters_cff")
 
 #---- for P5 (online) DB access
 process.load("DQM.Integration.config.FrontierCondition_GT_cfi")
diff --git a/DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_cff.py b/DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_cff.py
index 37714b63571..94184f2792e 100644
--- a/DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_cff.py
+++ b/DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_cff.py
@@ -6,11 +6,12 @@ from RecoPixelVertexing.PixelLowPtUtilities.siPixelClusterShapeCache_cfi import
 from DQM.SiPixelMonitorTrack.RefitterForPixelDQM import *
 
 hltSiPixelClusterShapeCache = siPixelClusterShapeCache.clone(src = 'hltSiPixelClusters')
-hltrefittedForPixelDQM = refittedForPixelDQM.clone(src ='hltMergedTracks')
+hltrefittedForPixelDQM = refittedForPixelDQM.clone(src ='hltMergedTracks',
+                                                   TTRHBuilder = cms.string('WithTrackAngle')) # no templates at HLT
 
 sipixelMonitorHLTsequence = cms.Sequence(
     hltSiPixelClusterShapeCache
     + hltSiPixelPhase1ClustersAnalyzer
-    + refittedForPixelDQM
+    + hltrefittedForPixelDQM
     + hltSiPixelPhase1TrackClustersAnalyzer
 )    
diff --git a/RecoPixelVertexing/PixelLowPtUtilities/plugins/SiPixelClusterShapeCacheProducer.cc b/RecoPixelVertexing/PixelLowPtUtilities/plugins/SiPixelClusterShapeCacheProducer.cc
index dba9894b23c..6322f9cc868 100644
--- a/RecoPixelVertexing/PixelLowPtUtilities/plugins/SiPixelClusterShapeCacheProducer.cc
+++ b/RecoPixelVertexing/PixelLowPtUtilities/plugins/SiPixelClusterShapeCacheProducer.cc
@@ -61,6 +61,13 @@ void SiPixelClusterShapeCacheProducer::produce(edm::StreamID, edm::Event& iEvent
   edm::Handle<InputCollection> input;
   iEvent.getByToken(token_, input);
 
+  if (!input.isValid()) {
+    edm::LogError("siPixelClusterShapeCache") << "input pixel cluster collection is not valid!";
+    auto output = std::make_unique<SiPixelClusterShapeCache>();
+    iEvent.put(std::move(output));
+    return;
+  }
+
   const auto& geom = &iSetup.getData(geomToken_);
 
   auto output = std::make_unique<SiPixelClusterShapeCache>(input);

```

technically makes the test `scram b runtests_TestDQMOnlineClient-hlt_dqm_sourceclient use-ibeos` run fine, even though it's not actually testing anything (see https://github.com/cms-sw/cmssw/pull/40875#issuecomment-1443608258).
I suppose that this would also fix the problem with [140.54](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cc786b/30879/runTheMatrixINPUT-results/140.54_RunPA2013), though we probably want to run some sequence that doesn't   run the HLT monitoring instead, since there's no re-HLT in it.

File: DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_cff.py (hunk 1 above)
Comment (arossi83) [reply DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_cff.py]: Yes, It should. Just push a commit to fix it.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by emanueleusai** [comment]
+1

- see comments in the master branch of this PR for a breakdown and explanation of the DQM differences.

**Comment by perrotta** [comment]
Please implement here the fix mentioned in the master version of this PR in https://github.com/cms-sw/cmssw/pull/40873#pullrequestreview-1315564157 as soon as the PR with the fix in the master is prepared. so that by tomorrow at most we can merge everything

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by arossi83** [comment]
Test failing because of #40889. Backport of #40873 with the addition of #40891 (see last commit here) (both PR already merged in master)

**Comment by mmusich** [comment]
@emanueleusai @mandrenguyen @perrotta w.r.t last signature this PR just removed cms type specification in one configuration file. Can you consider this for inclusion in 13_0_0?

**Comment by mandrenguyen** [comment]
+reconstruction
Disregarding relval input failure

**Comment by emanueleusai** [comment]
+1

- differences in DQM understood (see master PR)

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
merge

**Pull request merged**